src/test/UserServiceTest.java

```diff
@@ -6,146 +6,174 @@
 import org.junit.Assert;
 import org.junit.Test;
 import org.mockito.Mockito;
+
 import java.util.Objects;
 
 import static org.mockito.Mockito.when;
 
 /**
  * Test class for UserService.java
  */
-public class UserServiceTest {
-    private Fixture fixture = new Fixture();
+public class UserServiceTest
+{
+  private Fixture fixture = new Fixture();
+
+  @Test
+  public void testAddUser_addsUser()
+  {
+    fixture.givenUserIsInitialized("Ahmed", 10, 0);
+    fixture.whenUserIsAdded();
+    fixture.thenCountOfUserIs(1);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testAddUser_ageIsNegative_doesNotAddUser_throwsException()
+  {
+    fixture.givenUserIsInitialized("Ahmed", -10, 0);
+    fixture.whenUserIsAdded();
+    fixture.thenCountOfUserIs(0);
+    fixture.thenUserServiceThrewIllegalArgumentException();
+  }
+
+  @Test
+  public void testRemoveUser_removesUser()
+  {
+    //GIVEN
+    fixture.givenUserIsInitialized("Ahmed", 10, 1);
+    fixture.whenUserIsAdded();  //to reduce redundancy in Fixture class
+    //WHEN
+    fixture.whenUserIsDeleted();
+    //THEN
+    fixture.thenCountOfUserIs(0);
+  }
+
+  @Test(expected = IndexOutOfBoundsException.class)
+  public void testRemoveUser_userDoesNotExist_doesNotRemoveUser_throwsException()
+  {
+    //GIVEN
+    fixture.givenUserIsInitialized("Ali", 10, 0); // User count=1
+    fixture.whenUserIsAdded();
+    fixture.givenUserIsReInitializedAfterAddition("Ahmed", 11, 1); // assigning same object to a new reference
+    //WHEN
+    fixture.whenUserIsDeleted();  // will look for Ahmed in usersArray, won't find it so won't delete, count remains 1
+    //THEN
+    fixture.thenCountOfUserIs(1);
+    fixture.thenUserServiceThrewIndexOutOfBoundException();
+  }
+
+  @Test
+  public void testEditUser_editsUserSuccessfully()
+  {
+    //GIVEN
+    fixture.givenUserIsInitialized("Ahmed", 10, 0);
+    fixture.whenUserIsAdded();
+    //WHEN
+    fixture.whenUserIsEdited("Ali", 20);
+    //THEN
+    fixture.thenCheckIfUserIsEditedSuccessfully();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testEditUser_enteredInvalidAttributes_doesNotEditUser_throwsException()
+  {
+    //GIVEN
+    fixture.givenUserIsInitialized("Ahmed", 10, 0);
+    fixture.whenUserIsAdded();
+    //WHEN
+    fixture.whenUserIsEdited("", 0);
+    //THEN
+    fixture.thenCheckIfUserIsEditedSuccessfully();
+    fixture.thenUserServiceThrewIllegalArgumentException();
+  }
+}
+
+class Fixture
```

Comment:
please make this class as private, to avoid being able to use by any other class in same package by mistake